blockchain/blockchain.go

```diff
@@ -1514,11 +1514,13 @@ func (bc *BlockChain) insertChain(chain types.Blocks) (int, []interface{}, []*ty
 		// Write the block to the chain and get the status.
 		status, err := bc.WriteBlockWithState(block, receipts, stateDB)
 		if err != nil {
-			if err == ErrKnownBlock {
+			if err != ErrKnownBlock {
+				return i, events, coalescedLogs, err
+			}
+			if bc.CurrentBlock().NumberU64() >= block.NumberU64() {
```

Comment:
We don't need check this. We can use old code.

Reply:
good point